atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -651,13 +675,158 @@ def set_build_image(self):
+
+        build_config_name = None
+        build_image = None
+        imagestream = None
+        current_buildimage = self.get_current_buildimage()
+        status = get_build_json().get("status", {})
+
+        # orchestrator platform is same as platform on which we want to built on,
+        # so we can use the same image
+        if platforms == set([orchestrator_platform]):
+            self.build_image_digests[orchestrator_platform] = current_buildimage
+            return
+
+        # BuildConfig exists
+        if 'config' in status:
+            config = status['config']
+            if config['kind'] == 'BuildConfig':
+                build_config_name = config['name']
+            else:
+                raise RuntimeError("Build config type isn't BuildConfig : %s" % config['kind'])
+
+        # get image build from build metadata, which is set for direct builds
+        # this is explicitly set by osbs-client, it isn't default OpenShift behaviour
+        else:
+            metadata = get_build_json().get("metadata")
+            if 'from' in metadata['annotations']:
+                scratch_from = json.loads(metadata['annotations']['from'])
+
+                try:
+                    build_image, imagestream = self.process_image_from(scratch_from)
+                except UnknownKindException:
+                    raise RuntimeError("Build annotation has unknown 'kind' %s" %
+                                       scratch_from['kind'])
+            else:
+                raise RuntimeError("Build wasn't created from BuildConfig and neither"
+                                   "has 'from' annotation, which is needed for specified arch")
+
+        osbs = get_openshift_session(self.workflow, self.openshift_fallback)
+
+        # inspect BuildConfig
+        if build_config_name:
+            try:
+                build_config = osbs.os.get_build_config(build_config_name)
+            except OsbsException:
+                raise RuntimeError("Build config not found : %s" % build_config_name)
+
+            try:
+                build_from = build_config['spec']['strategy']['customStrategy']['from']
+            except KeyError:
+                raise RuntimeError("BuildConfig object is malformed")
+
+            try:
+                build_image, imagestream = self.process_image_from(build_from)
+            except UnknownKindException:
+                raise RuntimeError("BuildConfig object has unknown 'kind' %s" % build_from['kind'])
+
+        # if imageStream is used
+        if imagestream:
+            try:
+                tag = osbs.get_image_stream_tag(imagestream).json()
+            except OsbsException:
+                raise RuntimeError("ImageStreamTag not found %s" % imagestream)
+
+            try:
+                tag_image = tag['image']['dockerImageReference']
+            except KeyError:
+                raise RuntimeError("ImageStreamTag is malformed %s" % imagestream)
+
+            if '@sha256:' in tag_image:
```

Comment:
Minor (no action): 'sha256:' is part of the digest and in the future a different algorithm may be used. However, let's hope ImageStreams gain manifest list support before then!